code/game/objects/items/weapons/storage/fancy.dm

```diff
@@ -234,39 +234,45 @@
 /obj/item/storage/fancy/cigarettes/update_icon_state()
 	icon_state = "[initial(icon_state)][length(contents)]"
 
-/obj/item/storage/fancy/cigarettes/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
+/obj/item/storage/fancy/cigarettes/attack(mob/living/carbon/M, mob/living/user)
 	if(!ismob(M))
 		return
 
-	if(istype(M) && M == user && user.zone_selected == "mouth" && length(contents) > 0 && !user.wear_mask)
-		var/got_cig = 0
+	if(istype(M) && user.zone_selected == "mouth" && length(contents) > 0 && !M.wear_mask)
+		var/got_cig = FALSE
 		for(var/num=1, num <= length(contents), num++)
```

Comment:
```suggestion
		for(var/num in 1 to length(contents))
```
Test this to make sure it works the same as before

Reply:
It works. 👍